README.md

```diff
@@ -1 +1,113 @@
+
+[npm-url]:https://npmjs.org/package/@moxy/react-modal
+[downloads-image]:https://img.shields.io/npm/dm/@moxy/react-modal.svg
+[npm-image]:https://img.shields.io/npm/v/@moxy/react-modal.svg
+[travis-url]:https://travis-ci.org/moxystudio/react-modal
+[travis-image]:https://img.shields.io/travis/moxystudio/react-modal/master.svg
+[codecov-url]:https://codecov.io/gh/moxystudio/react-modal
+[codecov-image]:https://img.shields.io/codecov/c/github/moxystudio/react-modal/master.svg
+[david-dm-url]:https://david-dm.org/moxystudio/react-modal
+[david-dm-image]:https://img.shields.io/david/moxystudio/react-modal.svg
+[david-dm-dev-url]:https://david-dm.org/moxystudio/react-modal?type=dev
+[david-dm-dev-image]:https://img.shields.io/david/dev/moxystudio/react-modal.svg
+
+A modal, built on top of [react-modal](https://github.com/reactjs/react-modal), that also fixes the scrollable body in iOS.
+
+## Installation
+
+```sh
+$ npm install @moxy/react-modal
+```
+
+This library is written in modern JavaScript and is published in both CommonJS and ES module transpiled variants. If you target older browsers please make sure to transpile accordingly.
+
+
+## Usage
+
+First, make sure to bind modal to your appElement.
+
+```js
+import Modal from '@moxy/react-modal';
+
+Modal.setAppElement('#yourAppElement');
+```
+
+### Self controlled
+
+```js
+import React, { useRef, useCallback } from 'react';
+import Modal from '@moxy/react-modal';
+
+const MyComponent = () => {
+    const modalRef = useRef();
+
+    const handleModalOpen = useCallback(() => {
+        modalRef.current.open();
+    }, []);
+
+    return (
+        <>
+            <button onClick={ handleModalOpen }>Open</button>
+            <Modal>Modal content</Modal>
+        </>
+    );
+};
+
+export default MyComponent;
+```
+
+### Uncontrolled
+
+In this case, you have to close the modal when `onRequestClose` is called.
+
+```js
+import React, { useState, useCallback } from 'react';
+import Modal from '@moxy/react-modal';
+
+const MyComponent = () => {
+    const [isOpen, setOpen] = useState(false);
+
+    const handleModalOpen = useCallback(() => {
+        setOpen(true);
+    }, []);
+
+    const handleModalClose = useCallback(() => {
+        setOpen(false);
+    }, []);
+
+    return (
+        <>
+            <button onClick={ handleModalOpen }>Open</button>
+            <Modal onRequestClose={ handleModalClose }>Modal content</Modal>
+        </>
+    );
+};
+
+export default MyComponent;
+```
+
+## API
+
+### setAppElement
+
+Wrapper around [react-modal](https://github.com/reactjs/react-modal)'s `setAppElement` that not only bind the modal to the app element aswell as saves it to lock the scroll when a modal is open.
+
+### Modal
+
+React component that can be used as controlled or uncontrolled (check usage).
```

Comment:
Not sure if controlled/uncontrolled are the best descriptions. I'm assuming controls whether modals are self closing or closed by an external element. Since we're not looking for words that would be used in code (thus perhaps requiring some brevity at the expense of some clarity), i think we can be more verbose in our read me to guarantee that it's as clear as possible.

With that said, if it is indeed about self closing or nor, i suggest something like "If the modal can close itself" and "If the modal is closed from an external element".

Reply:
`controlled` and `uncontrolled` are the terms used by React itself: [Controlled Components](https://reactjs.org/docs/forms.html#controlled-components) and [Uncontrolled Components](https://reactjs.org/docs/uncontrolled-components.html).

I think we should use that convention since it's official.